packages/steiger-plugin-fsd/src/no-segments-on-sliced-layers/README.md

```diff
@@ -0,0 +1,47 @@
+# `no-segmentless-slices`
+
+Forbid segments that appear in direct children of sliced layers.
+
+Examples of project structures that pass this rule:
+
+```
+📂 shared
+  📂 ui
+    📄 index.ts
+  📂 lib
+    📄 index.ts
+📂 entities
+  📂 user
+    📂 ui
+    📂 model
+    📄 index.ts
+📂 pages
+  📂 home
+    📂 ui
+    📄 index.ts
+```
+
+Examples of project structures that fail this rule:
+
+```
+📂 shared
+  📂 ui
+    📄 index.ts
+  📂 lib
+    📄 index.ts
+📂 entities
+  📂 user
+    📂 ui
+    📂 model
+    📄 index.ts
+  📂 api // ❌
+    📄 index.ts
+📂 pages
+   📂 home
+     📂 ui
+     📄 index.ts
+```
+
+## Rationale
+
+Slices exist to partition code by business domain and entities. You can freely create and name them (e.g. `pages` home, profile and `entities` user, product, ...) based on your needs, application logic, company glossary, etc. Slices contain code of different type/purposes (segments) to implement their part of functionality. Segments (`ui`, `lib`, `api`) are simply a division of code by purpose, thus "ownerless" segments in sliced layers are not allowed since they need to be attached to some part of the business domain inside these layers.
```

Comment:
suggestion: I would extend this rationale a bit. You are describing the case when a folder `entities/api` was created as a segment, but there can also be cases when `entities/api` was meant to be a slice. Both cases are confusing, so we want to discourage them both. Here's my version of the rationale that describes both of these issues:
```suggestion
Several folder names like `ui` or `api` are conventionally understood in FSD as segments. When you have segments as direct children, it's either because that's the name you chose for your slice, or because some code ended up unsliced. In the first case, where that is the name of your slice, such a name is likely to cause confusion among developers who are familiar with FSD. In the second case, where a segment on a layer is because there was no slice to put it in, it's a violation of FSD structure, which decreases the benefits of FSD and also causes confusion.
```

Reply:
Yes, I didn't foresee that the user could choose api as the name of a slice